api-docs/src/main/kotlin/org/bitcoindevkit/bdk.kt

```diff
@@ -0,0 +1,453 @@
+package org.bitcoindevkit
+
+/**
+ * The cryptocurrency to act on.
+ */
+enum class Network {
+    /** Bitcoin's mainnet */
+    BITCOIN,
+
+    /** Bitcoin’s testnet */
+    TESTNET,
+
+    /** Bitcoin’s signet */
+    SIGNET,
+
+    /** Bitcoin’s regtest */
+    REGTEST,
+}
+
+/**
+ * A derived address and the index it was found at.
+ */
+data class AddressInfo (
+    /** Child index of this address. */
+    var index: UInt,
+
+    /** Address. */
+    var address: String
+)
+
+/**
+ * The address index selection strategy to use to derive an address from the wallet’s external descriptor.
+ *
+ * If you’re unsure which one to use, use `AddressIndex.NEW`.
+ */
+enum class AddressIndex {
+    /** Return a new address after incrementing the current descriptor index. */
+    NEW,
+
+    /** Return the address for the current descriptor index if it has not been used in a received transaction. Otherwise return a new address as with `AddressIndex.NEW`. Use with caution, if the wallet has not yet detected an address has been used it could return an already used address. This function is primarily meant for situations where the caller is untrusted; for example when deriving donation addresses on-demand for a public web page. */
```

Comment:
Another nit, but I'd wrap this into multiple lines.

Reply:
I brought all the long lines back, but didn't apply any particular rule. Do you have one in mind that you know is standard (length of line I mean?). I'm happy to apply a bit more consistently if there is a commonly agreed way to do KDocs/Javadoc documentation.